File: src/main/kotlin/com/tree/hometask1/task1/EdoDecisionTreeService.java
package com.tree.hometask1.task1;/*

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class EdoDecisionTreeService {

    @Autowired
    private EdoDecisionTreeRepo repo;


    public EdoDecisionTree getData(){

        List<EdoDecisionTree> data=repo.findAll();

        EdoDecisionTree parentNode=data.stream().filter(x->x.getParentId()==0).collect(Collectors.toList()).get(0);
        data.stream().filter(x->x.getParentId()!=0).


    }


}
*/

File: src/main/kotlin/com/tree/hometask1/task1/EdoDecisionTreeService.kt
package com.tree.hometask1.task1

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class EdoDecisionTreeService {

    @Autowired
    private lateinit var repo: EdoDecisionTreeRepo

    fun getData():EdoDecisionTreeDto {

        var allData=repo.findAll()

        //Create Dto object without Child-Node.
        var dtos=allData.map { x->EdoDecisionTreeDto(x) }

        //converting dtoList into Map<Id,EdoDecisionTreeDto>
        var dtoMap=dtos.associateBy { x->x.id }

        dtos.forEach { x->
            var parent=dtoMap.get(x.parentId)
            parent?.addChild(x)

        }
        var parentNode=dtos.filter { x->x.parentId==null }.first()

        return parentNode
    }
}